src/context.ts

```diff
@@ -865,6 +868,41 @@ export class Context<U extends Deunionize<tg.Update> = tg.Update> {
     })
   }
 
+  /**
+   * @see https://core.telegram.org/bots/api#sendchataction
+   *
+   * Sends the sendChatAction request repeatedly, with a delay between requests,
+   * as long as the provided callback function is being processed.
+   *
+   * The sendChatAction errors should be ignored, because the goal is the actual long process completing and performing an action.
+   *
+   * @param action - chat action type.
+   * @param callback - a function to run along with the chat action.
+   * @param extra - extra parameters for sendChatAction.
+   * @param {number} [extra.intervalDuration=8000] - The duration (in milliseconds) between subsequent sendChatAction requests.
+   */
+  async persistentChatAction(
+    action: Shorthand<'sendChatAction'>[0],
+    callback: () => Promise<void>,
+    extra?: tt.ExtraSendChatAction & { intervalDuration?: number }
+  ) {
+    await this.sendChatAction(action, { ...extra })
+
+    const timer = setInterval(
+      async () =>
+        await this.sendChatAction(action, { ...extra }).catch((err) => {
```

Comment:
Nitpick: unnecessary `async` / `await`
```suggestion
      () =>
        this.sendChatAction(action, { ...extra }).catch((err) => {
```